File: DataSummitSolution/DataSummitUI/src/app/efModels/drawingFeature.ts
import { Point } from "../profileVersion/create/models/point";
import { Drawing } from "../drawings/models/drawing";

export class DrawingFeature {

	DrawingFeatureId: number;
	Vendor: string;
	Value: string;
	DrawingId: number;
	Left?: number;
	Top?: number;
	Width?: number;
	Height?: number;
	Feature?: number;
	Center?: number;
	Drawing: Drawing;
	Points: Point[];

	constructor(drawingFeatureId?: number, vendor?: string, value?: string,
		drawingId?: number, left?: number, top?: number,
		width?: number, height?: number, feature?: number,
		center?: number, drawing?: Drawing, points?: Point[])
	{
		this.DrawingFeatureId = drawingFeatureId || 0;
		this.Vendor = vendor || "";
		this.Value = value || "";
		this.DrawingId = drawingId || 0;
		this.Left = left || 0 || null;
		this.Top = top || 0 || null;
		this.Width = width || 0 || null;
		this.Height = height || 0 || null;
		this.Feature = feature || 0 || null;
		this.Center = center || 0 || null;
		this.Drawing = drawing;
		this.Points = points;
	}
}

File: DataSummitSolution/DataSummitFunctions/Models/DrawingFeature.cs
using DataSummitFunctions.Models;

namespace DataSummitFunctions
{
    public partial class DrawingFeature
    {
        public long DrawingFeatureId { get; set; }
        public string Value { get; set; }
        public long Left { get; set; }
        public long Top { get; set; }
        public long Width { get; set; }
        public long Height { get; set; }
        public long Center { get; set; }
        public double Rotation { get; set; }
        public FeatureType Feature { get; set; }
        public long DrawingId { get; set; }

        public virtual Drawing Drawing { get; set; }

        public DrawingFeature()
        { }

        public enum FeatureType
        {
            Text = 1,
            Symbol = 2
        }
    }
}
